query-engine/connector-test-kit-rs/query-engine-tests/tests/queries/batching/select_with_diff_key_types.rs

```diff
@@ -0,0 +1,180 @@
+use query_engine_tests::*;
+
+#[test_suite(schema(schema))]
+mod select_different_key_type {
+    use indoc::indoc;
+    use query_engine_tests::{Runner, TestResult};
+
+    fn schema() -> String {
+        let schema = indoc! {
+            r#"
+                model FloatEntity {
+                    Id Float @id
```

Comment:
```suggestion
                    #id(Id, Float, @id)
```

here and below.

This will automatically add the missing `@map("_id")` annotation on MongoDB and should fix the tests failing there.

Reply:
done 👍
there was one more issue - Decimal not being supported in Mongo, I've decided to split this test into multiple test modules, one per key type, so we more easily adjust what runs for what database